cmd/thanos/query.go

```diff
@@ -119,6 +122,13 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 			fileSD = file.NewDiscovery(conf, logger)
 		}
 
+		queryMaxDuration := time.Duration(*queryTimeout)
+		storeReadMaxDuration := time.Duration(*storeReadTimeout)
+
+		if storeReadMaxDuration > queryMaxDuration {
```

Comment:
Depending on implementation... This might be not necesary, right?

Reply:
Its not necessary, right, because store context is child of query context. I added this check just to be consistent and clear.
But we can remove it without any side-effects.